### PR TITLE
Use suggestion for redundant_closure

### PR DESCRIPTION
**Pull request by sanxiyn**
cc #442.

### CODE REVIEW

**Comment by Manishearth** [comment]
```
tests/compile-fail/eta.rs:15:27: 15:44 error: redundant closure found
tests/compile-fail/eta.rs:15     let c = Some(1u8).map(|a| {1+2; foo}(a));
                                                       ^~~~~~~~~~~~~~~~~
tests/compile-fail/eta.rs:4:9: 4:26 note: lint level defined here
tests/compile-fail/eta.rs:4 #![deny(redundant_closure)]
                                    ^~~~~~~~~~~~~~~~~
tests/compile-fail/eta.rs:15:27: 15:44 help: for further information visit https://github.com/Manishearth/rust-clippy/wiki#redundant_closure
tests/compile-fail/eta.rs:15:27: 15:44 help: remove closure as shown:
tests/compile-fail/eta.rs:       let c = Some(1u8).map({1+2; foo});
```

The last suggestion has an extra block.

Also, the help should be shown before the "for further information". Perhaps we should have a `span_lint_and_then` similar to `span_lint_and_note` which runs a closure at the correct time if the lint level is correct.

**Comment by sanxiyn** [comment]
Eh, an extra block is a pre-existing problem.

**Comment by Manishearth** [comment]
Wait, nvm, we need that anyway. Sorry about the confusion. Okay, could you handle the build failure and the ordering of the suggestion and wiki link?

**Comment by sanxiyn** [comment]
Done.

File: src/utils.rs [hunk 1]
```diff
@@ -320,6 +325,17 @@ pub fn span_note_and_lint<T: LintContext>(cx: &T, lint: &'static Lint, span: Spa
     }
 }
 
+pub fn span_lint_and_then<T: LintContext, F>(cx: &T, lint: &'static Lint, sp: Span,
+        msg: &str, f: F) where F: Fn() {
+    cx.span_lint(lint, sp, msg);
+    f();
```
Comment (Manishearth) [inline src/utils.rs]: f() should only be run if the lint level is correct :)

Comment (sanxiyn) [inline src/utils.rs]: Oh. Fixed.

**Comment by Manishearth** [comment]
r+ pending travis

**Pull request merged**